cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/repositories.proto

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021-2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+package kubeappsapis.core.packages.v1alpha1;
+option go_package = "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1";
+
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+import "kubeappsapis/core/packages/v1alpha1/packages.proto";
+import "kubeappsapis/core/plugins/v1alpha1/plugins.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+// This protobuf file defines the request and response types for API calls
+// interacting with repositories, as well as the RepositoriesService to be implemented.
+
+// Each repositories v1alpha1 plugin must implement at least the following rpcs:
+//
+service RepositoriesService {
+  rpc AddPackageRepository(AddPackageRepositoryRequest) returns (AddPackageRepositoryResponse) {
+    option (google.api.http) = {
+      post: "/core/packages/v1alpha1/repositories"
+      body: "*"
+    };
+   }
+}
+
+// Standard request and response messages for each required function are defined
+// below for import and re-use.
+
+// -- Start definitions of the request messages --
+
+// AddPackageRepositoryRequest
+//
+// Request for AddPackageRepository
+message AddPackageRepositoryRequest {
+  // The target context where the package repository is intended to be 
+  // installed.
+  Context context = 1;
+
+  // A user-provided name for the package repository (e.g. bitnami)
+  string name = 2;
+
+  // A user-provided description. Optional
+  string description = 3;
+
+  // Whether this repository is global or namespace-scoped. Optional.
+  // By default, the value is false, i.e. the repository is global
+  bool namespace_scoped = 4;
```

Comment:
Currently in UI there is no such flag.
If the repo is created in the context of the global namespace, then it will be global. Why not using the same convention here? It might lead to confusion, e.g. repo is created with `namespace_scoped=false` but namespace in context is other than the global one. I guess logic in controller should throw some kind of error when `context` and `namespace_scoped` have inconsistent values?

Reply:
> Currently in UI there is no such flag. If the repo is created in the context of the global namespace, then it will be global. Why not using the same convention here?

I'll do my best to answer for Greg in his absence (PTO this week): Helm (and Carvel, for that matter) have the concept of a "global namespace", but it's not something that Flux has at all AIUI. So I think Greg is keen to design the API not to match the current UX assumptions necessarily, while still enabling the current functionality.

> It might lead to confusion, e.g. repo is created with `namespace_scoped=false` but namespace in context is other than the global one.

We (the kubeapps dashboard authors) can choose how to use this field (ie. not expose it to the user at all, but set it appropriately based on the namespace in context etc.) I think?

> I guess logic in controller should throw some kind of error when `context` and `namespace_scoped` have inconsistent values?

Yes, the helm plugin can handle this to match its expectations. Note, you can read more about the [decisions and questions for this API in the design doc](https://docs.google.com/document/d/1z03msZRGsRvcRaom-yrvEwIWcEfNy6fSA5Zg28gjYvA/edit#heading=h.u3lj7whdof7t).

Reply:
Couldn't have put it better myself. Thanks Michael

Reply:
ok, understood. Thanks!